src/Generating/FinishGen.cpp

```diff
@@ -1395,3 +1395,369 @@ eMonsterType cFinishGenPassiveMobs::GetRandomMob(cChunkDesc & a_ChunkDesc)
+		{E_BLOCK_DIAMOND_ORE,  0,        15,        1,        7},
+		{E_BLOCK_LAPIS_ORE,    0,        30,        1,        6},
+	};
+	return res;
+}
+
+
+
+
+
+const cFinishGenOres::OreInfos & cFinishGenOres::DefaultNetherOres(void)
+{
+	static OreInfos res;
+	if (res.empty())
+	{
+		// Hasn't been initialized yet, emplace the defaults:
+		//               OreType,                   OreMeta, MaxHeight, NumNests, NestSize
+		res.emplace_back(E_BLOCK_NETHER_QUARTZ_ORE, 0,       127,       20,       8);
+	}
+	return res;
+}
+
+
+
+
+
+const cFinishGenOres::OreInfos & cFinishGenOres::DefaultNaturalPatches(void)
+{
+	static OreInfos res;
+	if (res.empty())
+	{
+		// Hasn't been initialized yet, emplace the defaults:
+		//               OreType,        OreMeta,               MaxHeight, NumNests, NestSize
+		res.emplace_back(E_BLOCK_DIRT,   0,                     127,       20,       32);
+		res.emplace_back(E_BLOCK_GRAVEL, 0,                     127,       10,       32);
+		res.emplace_back(E_BLOCK_STONE,  E_META_STONE_GRANITE,  127,       20,       32);
+		res.emplace_back(E_BLOCK_STONE,  E_META_STONE_DIORITE,  127,       20,       32);
+		res.emplace_back(E_BLOCK_STONE,  E_META_STONE_ANDESITE, 127,       20,       32);
+	}
+	return res;
+}
+
+
+
+
+
+////////////////////////////////////////////////////////////////////////////////
+// cFinishGenOreNests:
+
+void cFinishGenOreNests::GenerateOre(
+	cChunkDesc & a_ChunkDesc,
+	BLOCKTYPE a_OreType, NIBBLETYPE a_OreMeta,
+	int a_MaxHeight, int a_NumNests, int a_NestSize,
+	int a_Seq
+)
+{
+	// This function generates several "nests" of ore, each nest consisting of number of ore blocks relatively adjacent to each other.
+	// It does so by making a random XYZ walk and adding ore along the way in cuboids of different (random) sizes
+	// Only "terraformable" blocks get replaced with ore, all other blocks stay (so the nest can actually be smaller than specified).
+
+	auto chunkX = a_ChunkDesc.GetChunkX();
+	auto chunkZ = a_ChunkDesc.GetChunkZ();
+	auto & blockTypes = a_ChunkDesc.GetBlockTypes();
+	auto & blockMetas = a_ChunkDesc.GetBlockMetasUncompressed();
+	for (int i = 0; i < a_NumNests; i++)
+	{
+		int nestRnd = m_Noise.IntNoise3DInt(chunkX + i, a_Seq, chunkZ + 64 * i) / 8;
+		int BaseX = nestRnd % cChunkDef::Width;
+		nestRnd /= cChunkDef::Width;
+		int BaseZ = nestRnd % cChunkDef::Width;
+		nestRnd /= cChunkDef::Width;
+		int BaseY = nestRnd % a_MaxHeight;
+		nestRnd /= a_MaxHeight;
+		int NestSize = a_NestSize + (nestRnd % (a_NestSize / 4));  // The actual nest size may be up to 1 / 4 larger
+		int Num = 0;
+		while (Num < NestSize)
+		{
+			// Put a cuboid around [BaseX, BaseY, BaseZ]
+			int rnd = m_Noise.IntNoise3DInt(chunkX + 64 * i, 2 * a_Seq + Num, chunkZ + 32 * i) / 8;
+			int xsize = rnd % 2;
+			int ysize = (rnd / 4) % 2;
+			int zsize = (rnd / 16) % 2;
+			rnd >>= 8;
+			for (int x = xsize; x >= 0; --x)
+			{
+				int BlockX = BaseX + x;
+				if ((BlockX < 0) || (BlockX >= cChunkDef::Width))
```

Comment:
I thought about implementing that a few days ago but I didn't find any occurrence of code where such a check would be useful